compiler/rustc_mir_build/src/thir/pattern/check_match.rs

```diff
@@ -948,10 +948,15 @@ fn report_non_exhaustive_match<'p, 'tcx>(
     err.note(format!("the matched value is of type `{}`", scrut_ty));
 
     if !is_empty_match {
-        let mut non_exhaustive_tys = FxHashSet::default();
+        let mut non_exhaustive_tys = FxIndexSet::default();
         // Look at the first witness.
         collect_non_exhaustive_tys(cx, &witnesses[0], &mut non_exhaustive_tys);
 
+        // Make order of diagnostics deterministic.
+        non_exhaustive_tys.sort();
```

Comment:
FxIndexSet will maintain its insertion order as the iteration order. Since it is filled in a deterministic way, we don't need to do any sorting (and also don't need to `#[allow]` the lint below).

Reply:
Oops, good catch, thanks. Done.